### PR TITLE
api: relax inline_string length limitation in DataSource

### PR DESCRIPTION
**Pull request by lizan**
As discussed in #14276, relax the min length requirement of  `inline_string` in DataSource. It should checked after DataSource since an empty file could pass validation.

Risk Level: Low
Testing: integration test
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: source/common/config/datasource.cc [hunk 1]
```diff
@@ -15,20 +15,27 @@ static constexpr uint32_t RetryCount = 1;
...
+    data = source.inline_string();
+    break;
   default:
     if (!allow_empty) {
       throw EnvoyException(
           fmt::format("Unexpected DataSource::specifier_case(): {}", source.specifier_case()));
     }
-    return "";
   }
+  if (!allow_empty && data.empty()) {
```
Comment (esmet) [inline source/common/config/datasource.cc:35]: Will this start enforcing allow_empty on the `kFilename` case where it didn't previously?

Comment (lizan) [reply source/common/config/datasource.cc:35]: Yes and this is intended in that way. I took quick look on existing use and mostly are allowing empty so this shouldn't be a big issue.

**Review by esmet — DISMISSED** [review]
Looks good

File: api/envoy/config/core/v3/base.proto [hunk 2]
```diff
@@ -334,7 +334,7 @@ message DataSource {
     bytes inline_bytes = 2 [(validate.rules).bytes = {min_len: 1}];
 
     // String inlined in the configuration.
-    string inline_string = 3 [(validate.rules).string = {min_len: 1}];
+    string inline_string = 3;
```
Comment (htuch) [inline api/envoy/config/core/v3/base.proto:337]: Should we also relax on `inline_bytes`?

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**